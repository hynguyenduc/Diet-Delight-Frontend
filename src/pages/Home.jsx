import React from 'react';
import NavBar from '../components/NavBar';
import Footer from '../components/Footer.jsx';
import HeroBanner from '../components/Herobanner';
import RecipeDisplay from '../components/HomePreview';

const Home = () => {
    return (
        <section className="hero is-fullheight" style={{
            minHeight: '80vh'
        }}>
            <div className="hero-head">
                <NavBar />
            </div>

            <div className="hero-body" style={{
                padding: '0'
            }}>
                <HeroBanner backgroundImage="https://upload.wikimedia.org/wikipedia/commons/1/10/Red_Color.jpg" />
            </div>

            <div className="featured-recipes" style={{ 
                paddingLeft: '2vw', paddingRight: '2vw', paddingBottom: '5vh'
            }}>
                <RecipeDisplay />
            </div>
            <Footer />
        </section>
    );
};

export default Home;
