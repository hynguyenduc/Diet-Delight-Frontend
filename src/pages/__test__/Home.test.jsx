// src/pages/__test__/Home.test.jsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import '@testing-library/jest-dom'; // Import jest-dom for additional matchers
import Home from '../Home';

// Mocking the HeroBanner and FeaturedRecipes components
vi.mock('../../components/Herobanner', () => ({
    default: () => <div>HeroBanner Component</div>
}));
vi.mock('../../components/FeaturedRecipes', () => ({
    default: () => <div>FeaturedRecipes Component</div>
}));

describe('Home Component', () => {
    it('renders the HeroBanner component', () => {
        render(<Home />);
        expect(screen.getByText('HeroBanner Component')).toBeInTheDocument();
    });

    it('renders the FeaturedRecipes component', () => {
        render(<Home />);
        expect(screen.getByText('FeaturedRecipes Component')).toBeInTheDocument();
    });

    it('should match the snapshot', () => {
        const { asFragment } = render(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
});
