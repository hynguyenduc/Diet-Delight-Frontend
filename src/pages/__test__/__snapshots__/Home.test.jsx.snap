// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="hero is-fullheight"
  >
    <div
      id="hero-body"
    >
      <div>
        HeroBanner Component
      </div>
    </div>
    <div
      class="featured-recipes"
      id="featuredrecipes"
    >
      <div>
        FeaturedRecipes Component
      </div>
    </div>
  </section>
</DocumentFragment>
`;
